day_3/story_game.py

```diff
@@ -31,17 +33,20 @@ def make_story(self):
                 text=details['text'],
             )
             self.story[position] = node
-        event_codes = self.story.keys()
+        event_codes = set(self.story.keys())
+        event_codes.add(None)
+        if event_codes != all_positions:
```

Comment:
_nit_: it took some time to understand what's going on here... maybe it'd be better to rename `all_positions` &rarr; `choices_codes` and `event_codes` &rarr; `nodes_codes` or something like that, so it's clear that we're making sure that all the codes, that are used as a choice options, are actually presented in the story codes. 

Although I think that `!=` is too much of a check. Probably `issubset` will be better here: you may not use all the nodes from the story, but you must be sure that everything that you use is presented.

Reply:
> Although I think that != is too much of a check. Probably issubset will be better here: you may not use all the nodes from the story, but you must be sure that everything that you use is presented.
As for me, I think it would be a better check that there is no extra information in the scenario.